.github/workflows/CI.yml

```diff
@@ -11,7 +11,7 @@ name: continuous-integration
 # Repeated pushes to a PR will cancel all previous builds, while multiple
 # merges to main will not cancel.
 concurrency:
-  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}
+  group: ci-${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}
```

Comment:
This avoids the race condition we currently see in https://github.com/pydata/pydata-sphinx-theme/actions/runs/13914118007

since we are calling this workflow from `publsh.yml` the current concurrency definition means both workflows will have the same group, and thus will cancel each other.